### PR TITLE
Move data fetching to cron jobs / redis

### PR DESCRIPTION
**Pull request by koffeinfrei**
Instead of fetching all the data when requesting the data, we move all the fetching to cron jobs. This way the fetching is done independently of the web requests, and the web applications' boot time is significantly faster.

To store the data we redis. The dashboard reads all its data from redis now. No direct access to third party services should be performed anymore from the dashboard jobs.

In development mode, the data is still fetched at boot time.

### CODE REVIEW

**Comment by koffeinfrei** [comment]
Ready for review.

**Comment by huerlisi** [comment]
:+1:

File: jobs/controllr.rb [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
   # salaries per month
-  points = YAML.load_file('config/salaries.yml')[Date.today.year].map { |key, value| { x: key, y: value } }
-  send_event('salary-graph', points: points)
+  send_event('salary-graph', DataStore.get('salary-graph'))
 end
 
 update
 
-SCHEDULER.every '1d' do
+SCHEDULER.every '10m' do
```
Comment (exul) [inline jobs/controllr.rb:25]: Why down to 10 minutes? I think 1 day was totally fine for controllr data, because they don't change that frequently.

**Comment by exul** [comment]
:+1:

File: jobs/controllr.rb (hunk 1 above)
Comment (koffeinfrei) [inline jobs/controllr.rb:25]: The API calls are in the cron job (rake task) anyway, where it's still 1 day. This is just the poll frequency to redis from the dashboard, which I unified to 10min for all jobs.
Having 1 day for the dashboard poll and 1 day for the cron job frequency, in the worst case the current data shown could be almost 2 days old.

Comment (exul) [inline jobs/controllr.rb:25]: ah, I see, thanks

**Pull request merged**